std/json.d

```diff
@@ -346,10 +346,11 @@ struct JSONValue
             type_tag = JSON_TYPE.STRING;
             store.str = arg;
         }
-        else static if(is(T : char[]))
+        else static if(is(T : const char[]) || is(T : const wchar[]) || is(T : const dchar[]))
         {
+            import std.utf : byUTF;
             type_tag = JSON_TYPE.STRING;
-            store.str = arg.idup;
+            store.str = byUTF!char(arg).array.idup;
```

Comment:
@lionello Unfortunately, it looks like we would have to resort to the trusted lambda hack:
```
else static if (is(T : E[], E) && isSomeChar!E)
{
    import std.utf : byUTF;    
    () @trusted {
        import std.exception : assumeUnique;
        store.str = arg.byUTF!char.array.assumeUnique;
    }();
    type_tag = JSON_TYPE.STRING;
}
```
Even though, `array` is correctly deduced as pure:
```
char[] k = ['a', 's', 'd'];
auto r = k.byUTF!char;
string s = array(r).assumeUnique;	
pragma (msg, typeof(array!(typeof(r))));
// prints: pure nothrow @safe char[](ByCodeUnitImpl r)
```